### PR TITLE
Re-enable shorthand HTTP API

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What changed?
This re-enables the shorthand payload format for our HTTP API.

## Why?
This is a business requirement for our HTTP API.

## How did you test it?
I updated our HTTP API tests to test all four mimetypes we support

## Potential risks
None

## Is hotfix candidate?
No.

### CODE REVIEW

File: service/frontend/protojson_marshaler.go [hunk 1]
```diff
@@ -30,39 +30,77 @@ import (
...
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
-	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
+
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/api/temporalproto"
+)
+
+const (
+	mimeJSON                  = "application/json"
```
Comment (cretz) [inline service/frontend/protojson_marshaler.go:40]:
```suggestion
	mimeAll                   = "*"
```

I think we should support unset mime type to be JSON too. I just did this simply as "any unrecognized mime type" but if you want to be more strict when user does specify it, I suppose we can though probably no need.

File: service/frontend/protojson_marshaler.go [hunk 2]
```diff
@@ -30,39 +30,77 @@ import (
...
+		uOpts: temporalproto.CustomJSONUnmarshalOptions{
+			Metadata: metadata,
 		},
 	}
 }
 
-func (p protoJsonMarshaler) Marshal(v any) ([]byte, error) {
+var logfunc = func(string, ...any) {}
+
+func Setlogfunc(f func(string, ...any)) { logfunc = f }
```
[the remark this answers is not in the document]
Comment (tdeebswihart) [reply service/frontend/protojson_marshaler.go:99]: Nope. I once again left debugging in my PR…

File: service/frontend/http_api_server.go
Comment (bergundy) [inline service/frontend/http_api_server.go:1]: nit: I still see a couple of references to gogo in comments in this file.

File: service/frontend/protojson_marshaler.go [hunk 3]
```diff
@@ -30,38 +30,70 @@ import (
...
+	commonpb "go.temporal.io/api/common/v1"
+	"go.temporal.io/api/temporalproto"
+)
+
+const (
+	mimeAll                   = "*"
+	mimePrettyJSONNoShorthand = "application/json+pretty+no-payload-shorthand"
+	mimeJSONNoShorthand       = "application/json+no-payload-shorthand"
+	mimePrettyJSONShorthand   = "application/json+pretty"
 )
```
Comment (bergundy) [inline service/frontend/protojson_marshaler.go:44]: nit: I know we do this all over the codebase, but I prefer just having the value in line and avoid the extra lookup.

**Review by bergundy — APPROVED** [review]
I only had a couple of nits.

Not super familiar with this part of the codebase but the changes LGTM.

File: service/frontend/http_api_server.go
Comment (tdeebswihart) [reply service/frontend/http_api_server.go:1]: I don't have enough context to know whether the claims there are still true. @cretz, as the author, do the claims about gogo-status-compatible but not grpc-status-compatible in errorHandler still hold?

File: service/frontend/protojson_marshaler.go (hunk 3 above)
Comment (tdeebswihart) [reply service/frontend/protojson_marshaler.go]: I inlined them

**Comment by tdeebswihart** [comment]
I'm setting this to automerge when tests pass to unblock other work that's hurt by the api-go changes. This should be the last time I break things

**Pull request merged**